pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -579,6 +579,137 @@ def to_abstract_repr(self) -> str:
         validate_abstract_repr(abstr_dev_str, "device")
         return abstr_dev_str
 
+    def print_specs(self) -> None:
+        """Prints the device specifications."""
+        title = f"{self.name} Specifications"
+        header = ["-" * len(title), title, "-" * len(title)]
+        print("\n".join(header))
+        print(self._specs())
+
+    @property
+    def specs(self) -> str:
+        """Text summarizing the specifications of the device."""
+        return self._specs(for_docs=False)
+
+    def _specs(self, for_docs: bool = False) -> str:
+        def yes_no_fn(cond: bool) -> str:
+            return "Yes" if cond else "No"
+
+        lines = [
+            "\nRegister parameters:",
+            f" - Dimensions: {self.dimensions}D",
+            f" - Rydberg level: {self.rydberg_level}",
+            f" - Maximum number of atoms: {self.max_atom_num}",
+            f" - Maximum distance from origin: {self.max_radial_distance} μm",
```

Comment:
These parameters can be `None` in a `VirtualDevice`. 
I think saying `"Maximum number of atoms: None"` is acceptable (albeit still potentially confusing), but `"None μm" should be avoided.

Reply:
I have been hesitating on this point, because it seems useful to indicate that there are no limits. For this reason, when I was working on the code, I did hesitate to still add a line for any field being `None`, to indicate that there is no constraint. On the other hand, this may be confusing as you say, so i decided to follow the current convention of removing the lines altogether. I did not know that these values could also be `None`, so I suggest (at least for now) to also remove the lines. But I think it would be worth having other opnions on the matter.

Reply:
There was a slight inconsistency with what I said, since I had left `max_layout_traps`. Now I am also removing the line if it's `None`.

Reply:
I agree that showing there is no constraint can be useful but let's stick with not showing it for now.